sound/soc/sof/pcm.c

```diff
@@ -711,6 +742,8 @@ static int sof_pcm_probe(struct snd_soc_component *component)
 		return ret;
 	}
 
+	sof_set_longname(component);
```

Comment:
I wonder if combination of "DMI name + SOF-tag + topology file name" would be sufficient. This would allow to match for a specific product and export the dynamic detection results of how SOF chooses the topology file.